app/services/migrations/remove_extra_derivatives.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Migrations
+  class RemoveExtraDerivatives
+    def self.call
+      new.run
+    end
+
+    def run
+      progress_bar
+      query_service.find_all_of_model(model: FileSet).each do |file_set|
+        progress_bar.progress += 1
+        next unless extra_derivatives?(file_set) || extra_thumbnails?(file_set)
+
+        RegenerateDerivativesJob.perform_later(file_set.id.to_s)
+      end
+    end
+
+    private
+
+      def progress_bar
+        @progress_bar ||= ProgressBar.create format: "%a %e %P% Resources Processed: %c of %C", total: query_service.custom_queries.count_all_of_model(model: FileSet)
+      end
+
+      def extra_derivatives?(file_set)
+        file_set.file_metadata.count(&:derivative?) > 1
```

Comment:
It's looking like > 2 is a decent check, I'm running a longer query to see if there's any that we wouldn't want to regenerate that have that count.

Reply:
we could try limiting it to file sets on raster resources, the bug was in the raster resource derivative service